src/data_morph/shapes/points.py

```diff
@@ -304,3 +305,51 @@ def distance(self, x: Number, y: Number) -> int:
             Always returns 0 to allow for scattering of the points.
         """
         return 0
+
+
+class Spiral(PointCollection):
+    """
+    Class for the spiral shape.
+
+    .. plot::
+       :scale: 100
+       :caption:
+            This shape is generated using the panda dataset.
+
+        from data_morph.data.loader import DataLoader
+        from data_morph.shapes.points import Spiral
+
+        _ = Spiral(DataLoader.load_dataset('panda')).plot()
+
+    Parameters
+    ----------
+    dataset : Dataset
+        The starting dataset to morph into other shapes.
+
+    Notes
+    -----
+    The formula for a spiral can be found here:
+    https://en.wikipedia.org/wiki/Archimedean_spiral
+    """
+
+    def __init__(self, dataset: Dataset) -> None:
+        xmin, xmax = dataset.data_bounds.x_bounds
+        ymin, ymax = dataset.data_bounds.y_bounds
+
+        # Coordinates of centre
+        cx = dataset.df.x.mean()
+        cy = dataset.df.y.mean()
+
+        # Max radius
+        radius = min(xmax - xmin, ymax - ymin) / 2
+
+        # Number of rotations
+        num_rotations = 3
+
+        t = np.linspace(0, 1, num=200)
```

Comment:
Two other thoughts here:

1. Can we use fewer points? This would speed up the distance calculations.
2. The points are farther apart as the spiral grows. Can we reduce the density at the center and fill in the outer parts better?